### ISSUE TITLE
Fatal Exception: java.lang.IllegalStateException Reply already submitted

### ISSUE THREAD

**Issue by chandrabezzo** [kineapps/flutter_file_dialog#31]
Hi team, from our Firebase Crashlytics we always got issue about `IllegalStateException` from your plugin. I think this crash happen because your plugin send `result.error` more than one submitted.

```
Fatal Exception: java.lang.IllegalStateException: Reply already submitted
       at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:35)
       at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:14)
       at com.kineapps.flutter_file_dialog.FileDialog$saveFileOnBackground$1.invokeSuspend(FileDialog.java:142)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:11)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:80)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:233)
       at android.os.Looper.loop(Looper.java:334)
       at android.app.ActivityThread.main(ActivityThread.java:8396)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:582)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1068)
```

Our Environment:
[✓] Flutter (Channel unknown, 2.8.1, on macOS 12.4 21F79 darwin-arm, locale en-ID)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.69.2)

Using specific version
```dart
flutter_file_dialog: ^2.3.0
```

**Comment by kinex** [comment]
Another similar crash:
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4940)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:4981)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2046)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:225)
  at android.app.ActivityThread.main (ActivityThread.java:7564)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:950)
Caused by: java.lang.IllegalStateException: 
  at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply (Unknown Source:35)
  at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success (Unknown Source:14)
  at com.kineapps.flutter_file_dialog.FileDialog.onActivityResult (FileDialog.java:161)
  at io.flutter.embedding.engine.FlutterEngineConnectionRegistry$FlutterEngineActivityPluginBinding.onActivityResult (Unknown Source:25)
  at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.onActivityResult (Unknown Source:13)
  at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult (Unknown Source:51)
  at io.flutter.embedding.android.FlutterActivity.onActivityResult (Unknown Source:10)
  at android.app.Activity.dispatchActivityResult (Activity.java:8274)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4933)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:4981)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2046)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:225)
  at android.app.ActivityThread.main (ActivityThread.java:7564)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:950)
```

**Comment by chandrabezzo** [comment]
Hi @kinex thanks for resolve this issue. But, when you will publish the new version with resolved this issue? I see you code fixed 6 days ago. But, I see in the last version is 9 months ago. For now, our team still use Flutter version under 3. So, please support under Flutter 3. Because I always found another plugin that only support Flutter 3 without concern backward version. Thanks.

**Comment by kinex** [comment]
Sorry for the delay. Fix is included in 2.3.1
